src/sia/models/sia_query_params.py

```diff
@@ -0,0 +1,411 @@
+        Query(
+            title="timeres",
+            description="Range(s) of temporal resolution",
+            examples=["-Inf 1.0"],
+        ),
+    ] = None
+
+    specrp: Annotated[
+        list[str] | None,
+        Query(
+            title="specrp",
+            description="Range(s) of spectral resolving power",
+            examples=["1000 2000"],
+        ),
+    ] = None
+
+    q_id: Annotated[
+        list[str] | None,
+        Query(
+            title="id",
+            alias="id",
+            description="Identifier of dataset(s)",
+            examples=["obs_id_1"],
+        ),
+    ] = None
+
+    dptype: Annotated[
+        list[DPType] | None,
+        Query(title="dptype", description="Type of data", examples=["image"]),
+    ] = None
+
+    calib: Annotated[
+        list[CalibLevel] | None,
+        Query(
+            title="calib",
+            description="Calibration level of the data",
+            examples=[0, 1, 2],
+        ),
+    ] = None
+
+    target: Annotated[
+        list[str] | None,
+        Query(
+            title="target", description="Name of the target", examples=["M31"]
+        ),
+    ] = None
+
+    collection: Annotated[
+        list[str] | None,
+        Query(
+            title="collection",
+            description="Name of the data collection",
+            examples=["HST"],
+        ),
+    ] = None
+
+    facility: Annotated[
+        list[str] | None,
+        Query(
+            title="facility",
+            description="Name of the facility",
+            examples=["HST"],
+        ),
+    ] = None
+
+    instrument: Annotated[
+        list[str] | None,
+        Query(
+            title="instrument",
+            description="Name of the instrument",
+            examples=["ACS"],
+        ),
+    ] = None
+
+    maxrec: Annotated[
+        int | None,
+        Query(
+            title="maxrec",
+            description="Maximum number of records in the response",
+            examples=[10],
+        ),
+    ] = None
+
+    responseformat: Annotated[
+        str | None,
+        Query(
+            title="reponseformat",
+            description="Format of the response",
+            examples=["application/x-votable+xml"],
+        ),
+    ] = None
+
+    @classmethod
+    def from_dict(cls, data: dict[str, Any]) -> "SIAQueryParams":
+        """Create an instance of SIAQueryParams from a dictionary.
+
+        This method handles renaming 'id' to 'q_id' and 'format' to 'q_format'.
+
+        Parameters
+        ----------
+        data
+            The dictionary containing the query parameters.
+
+        Returns
+        -------
+        SIAQueryParams
+            Instance of SIAQueryParams initialized with the provided data.
+        """
+        if "id" in data:
+            data["q_id"] = data.pop("id")
+
+        if "format" in data:
+            data["q_format"] = data.pop("format")
+
+        return cls(**data)
+
+    @classmethod
+    def validate_enum_list(
```

Comment:
This is more stuff that you'll be able to drop if you use the approach of two separate dependencies mentioned above, I think.

Reply:
I think you were suggesting dropping if using Pydantic right? I can't drop i think if its a dataclass?

Reply:
If you write a FastAPI dependency that defines a `list[SomeEnum]` type parameter with the `Query` annotation, I think it will just do the right thing and give you a validated list of enum objects in the body, which you can then use to construct the model.